### DISCUSSION TITLE
Support for XCTestCase.addUIInterruptionMonitor

### DISCUSSION THREAD

**Discussion by AlexTanabe** [Tyler-Keith-Thompson/CucumberSwift#49] [Ideas]
Hello,

I am integrating your solution in my project at the moment, and so far looks pretty nice.
However, before using CucumberSwift, we used plain XCTestCase and one of the functionalities we used was the `addUIInterruptionMonitor` method inside XCTestCase.

This was being used to handle system dialogs and dismiss them if needed inside our UI Tests...

Is there any way of including this functionality inside CucumberSwift, similar to what you did with expectations?

If not, is there any alternative you might suggest me?
Thanks in advance!

P.S: In the meantime, I am thinking of using the XCTestObservation protocol methods of `testCaseWillStart` and `testCaseDidFinish` to get a reference to the current test case being executed, and use that inside my step definitions to add/remove interruption monitors.

**Comment by Tyler-Keith-Thompson** [comment]
CucumberSwift steps have [an optional reference to their test case](https://tyler-keith-thompson.github.io/CucumberSwift/documentation/cucumberswift/step/testcase). You can call `addInterruptionMonitor` on that. 

Example:
```swift
Given("^some precondition$") { _, step in
    // because the step is executing when this closure is called, testCase should not be nil.
    step.testCase?.addInterruptionMonitor()
}
```

**Reply by badrinrs** [reply]
Can you add a test case reference to the scenario object as well? That would help in adding screenshot post test run in AfterScenario Hook and add use `addInterruptionMonitor` in BeforeScenario Hook.

**Reply by Tyler-Keith-Thompson** [reply]
The trick there is that a Scenario doesn't have a single test case, it has a test case per step...perhaps you could do `scenario.steps.last?.testCase?.addInterruptionMonitor`

**Reply by badrinrs** [reply]
Ok! But I feel Scenario should be analogous to XCTestCase and steps should be part of the XCTestCase.
